src/MISC/fix_eco_force.h

```diff
@@ -0,0 +1,91 @@
+ DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+ certain rights in this software.  This software is distributed under
+ the GNU General Public License.
+
+ See the README file in the top-level LAMMPS directory.
+ ------------------------------------------------------------------------- */
+
+#ifdef FIX_CLASS
+
+FixStyle(eco/force,FixECOForce)
+
+#else
+
+#ifndef LMP_FIX_ECO_FORCE_H
+#define LMP_FIX_ECO_FORCE_H
+
+#include "fix.h"
+
+namespace LAMMPS_NS {
+
+#define FIX_ECO_FORCE_MAX_NEIGHBORS 24
+
+class FixECOForce : public Fix {
+  public:
+    struct Nbr {
+      int n;                    // # of neighbors
+      tagint id[FIX_ECO_FORCE_MAX_NEIGHBORS];       // IDs of each neighbor
+                                // if center atom is owned, these are local IDs
+                                // if center atom is ghost, these are global IDs
+      double duchi;             // potential derivative
+      double delta[FIX_ECO_FORCE_MAX_NEIGHBORS][3]; // difference vectors
+      double real_phi[2][3];    // real part of wave function
+      double imag_phi[2][3];    // imaginary part of wave function
+    };
+
+    FixECOForce(class LAMMPS *, int, char **);
+    ~FixECOForce();
+    int setmask();
+    void init();
+    void init_list(int, class NeighList *);
+    void setup(int);
+    void post_force(int);
+    void post_force_respa(int, int, int);
+    double compute_scalar();
+    int pack_forward_comm(int, int *, double *, int, int *);
+    void unpack_forward_comm(int, int, double *);
+    double memory_usage();
+
+  private:
+    int me;                             // this processors rank
+    int nmax;                           // maximal # of owned + ghost atoms on this processor
+    int ilevel_respa;                   // used for RESPA integrator only
+
+    int sign;                           // from sign of u
+    double u_0;                         // synthetic potential energy
+    double half_u;                      // half synthetic potential energy
+    double eta;                         // threshold for thermal effects
+    double inv_eta;                     // inverse threshold for thermal effects
+    double r_cut;                       // cutoff radius
+    double squared_cutoff;              // squared cutoff radius
+    double inv_squared_cutoff;          // inverse squared cutoff radius
+    char *dir_filename;                 // filename of reference grain input
+    double dir_vec[6][3];               // direct lattice vectors
+    double reciprocal_vectors[2][3][3]; // reciprocal lattice vectors
+
+    double added_energy;        // energy added by fix
+
+    double **order;             // order parameter and normalized order
+                                // parameter per atom
+
+    double norm_fac;            // normalization constant
+    double inv_norm_fac;        // inverse normalization constant
+
+    Nbr *nbr;                   // pointer on array of Nbr structs
```

Comment:
please put just a forward declaration here and define this struct in the .cpp file only.